testapi.pm

```diff
@@ -588,7 +600,12 @@ sub script_output($;$) {
     $wait ||= 10;
 
     my $suffix = _random_string;
-    type_string "curl -f -v " . autoinst_url . "/current_script > /tmp/script$suffix.sh && echo \"curl-\$?\" > /dev/$serialdev\n";
+    if (get_var("CONNECT_PASSWORD")) {
+        type_string "curl -f -v '" . autoinst_url . "/current_script?connect_password=" . get_var("CONNECT_PASSWORD") . "' > /tmp/script$suffix.sh && echo \"curl-$?\" > /dev/$serialdev\n";
```

Comment:
I don't like this code duplication and I think autoinst_url can be extended to do that for you to begin with. You will need to pass a path then